azure-iot-provisioning-devicesdk/azure/iot/provisioning/devicesdk/sk_registration_client.py

```diff
@@ -0,0 +1,88 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""
+This module contains one of the implementations of the Registration Client which uses Symmetric Key authentication.
+"""
+import logging
+from threading import Event
+from .registration_client import RegistrationClient
+
+logger = logging.getLogger(__name__)
+
+
+class SymmetricKeyRegistrationClient(RegistrationClient):
+    """
+    Client which can be used to run the registration of a device using Symmetric Key authentication.
+    """
+
+    def __init__(self, transport):
+        """
+        Initializer for the Symmetric Key Registration Client
+        """
+        super(SymmetricKeyRegistrationClient, self).__init__(transport)
+        self._transport.on_transport_connected = self._on_state_change
+        self._transport.on_transport_disconnected = self._on_state_change
+        self._transport.on_transport_registration_update = self._on_device_registration_update
+
+    def register(self):
+        """
+        Register the device with the provisioning service.
+        """
+        logger.info("Sending message to Hub...")
+        subscribe_complete = Event()
+        send_request_complete = Event()
+
+        def callback_subscribe():
+            subscribe_complete.set()
+            logger.info("Successfully subscribed to Hub")
+
+        def callback_request():
+            send_request_complete.set()
+            logger.info("Successfully sent request to Hub")
+
+        self._transport.send_registration_request(
+            callback_subscribe=callback_subscribe, callback_request=callback_request
+        )
+        # TODO : Chnages to state machine inside mqtt
+        subscribe_complete.wait() and send_request_complete.wait()
+
+    def cancel(self):
+        """
+        Cancels the current registration process.
+        """
+        pass
+
+    def disconnect(self):
```

Comment:
>disconnect [](start = 8, length = 10)

What did we decide about including disconnect?  Does it make sense to have the disconnect be part of the registration flow?  in other words, does "register" do connect->register->disconnect like node does?

Reply:
Disconnect does not make sense to be a part of register. But as of now the sample is working with disconnect, so I will leave it for now. Since we are revisiting the state machine we can change disconnect to be part of one of the polling or request response and delete it from client. 

---
In reply to: [272690807](https://github.com/Azure/azure-iot-sdk-python-preview/pull/59#discussion_r272690807) [](ancestors = 272690807)